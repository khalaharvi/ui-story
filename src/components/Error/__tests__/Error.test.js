import React from 'react';
import { shallow } from 'enzyme';

import ErrorComponent from '../Error';

describe('<ErrorComponent />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ErrorComponent
        logo={<img src="http://placehold.it/150x30" alt="" />}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom className', () => {
    const wrapper = shallow(
      <ErrorComponent
        className="custom-classname"
        logo={<img src="http://placehold.it/150x30" alt="" />}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with link', () => {
    const wrapper = shallow(
      <ErrorComponent
        logo={<img src="http://placehold.it/150x30" alt="" />}
        link={<a href="/">Back to Homepage</a>}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom title', () => {
    const wrapper = shallow(
      <ErrorComponent
        logo={<img src="http://placehold.it/150x30" alt="" />}
        title="Custom Title"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom message', () => {
    const wrapper = shallow(
      <ErrorComponent
        logo={<img src="http://placehold.it/150x30" alt="" />}
        message="Custom Message"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with copyright', () => {
    const wrapper = shallow(
      <ErrorComponent
        logo={<img src="http://placehold.it/150x30" alt="" />}
        copyright="Custom Copyright, LLC"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
