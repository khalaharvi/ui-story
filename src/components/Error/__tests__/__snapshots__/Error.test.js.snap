// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorComponent /> renders correctly 1`] = `
<div
  className="dh-error"
>
  <header
    className="dh-error-header"
  >
    <div
      className="dh-error-logo"
    >
      <img
        alt=""
        src="http://placehold.it/150x30"
      />
    </div>
  </header>
  <section
    className="dh-error-section"
  >
    <div>
      <h1>
        Internal Server Error
      </h1>
      <p>
        We're sorry, but something went wrong. We have been notified about this and we'll take a look at it shortly.
      </p>
    </div>
  </section>
  <footer
    className="dh-error-footer"
  >
    <small>
      © 
      2018
       
      Dering Hall, Inc.
    </small>
  </footer>
</div>
`;

exports[`<ErrorComponent /> renders correctly with copyright 1`] = `
<div
  className="dh-error"
>
  <header
    className="dh-error-header"
  >
    <div
      className="dh-error-logo"
    >
      <img
        alt=""
        src="http://placehold.it/150x30"
      />
    </div>
  </header>
  <section
    className="dh-error-section"
  >
    <div>
      <h1>
        Internal Server Error
      </h1>
      <p>
        We're sorry, but something went wrong. We have been notified about this and we'll take a look at it shortly.
      </p>
    </div>
  </section>
  <footer
    className="dh-error-footer"
  >
    <small>
      © 
      2018
       
      Custom Copyright, LLC
    </small>
  </footer>
</div>
`;

exports[`<ErrorComponent /> renders correctly with custom className 1`] = `
<div
  className="dh-error custom-classname"
>
  <header
    className="dh-error-header"
  >
    <div
      className="dh-error-logo"
    >
      <img
        alt=""
        src="http://placehold.it/150x30"
      />
    </div>
  </header>
  <section
    className="dh-error-section"
  >
    <div>
      <h1>
        Internal Server Error
      </h1>
      <p>
        We're sorry, but something went wrong. We have been notified about this and we'll take a look at it shortly.
      </p>
    </div>
  </section>
  <footer
    className="dh-error-footer"
  >
    <small>
      © 
      2018
       
      Dering Hall, Inc.
    </small>
  </footer>
</div>
`;

exports[`<ErrorComponent /> renders correctly with custom message 1`] = `
<div
  className="dh-error"
>
  <header
    className="dh-error-header"
  >
    <div
      className="dh-error-logo"
    >
      <img
        alt=""
        src="http://placehold.it/150x30"
      />
    </div>
  </header>
  <section
    className="dh-error-section"
  >
    <div>
      <h1>
        Internal Server Error
      </h1>
      <p>
        Custom Message
      </p>
    </div>
  </section>
  <footer
    className="dh-error-footer"
  >
    <small>
      © 
      2018
       
      Dering Hall, Inc.
    </small>
  </footer>
</div>
`;

exports[`<ErrorComponent /> renders correctly with custom title 1`] = `
<div
  className="dh-error"
>
  <header
    className="dh-error-header"
  >
    <div
      className="dh-error-logo"
    >
      <img
        alt=""
        src="http://placehold.it/150x30"
      />
    </div>
  </header>
  <section
    className="dh-error-section"
  >
    <div>
      <h1>
        Custom Title
      </h1>
      <p>
        We're sorry, but something went wrong. We have been notified about this and we'll take a look at it shortly.
      </p>
    </div>
  </section>
  <footer
    className="dh-error-footer"
  >
    <small>
      © 
      2018
       
      Dering Hall, Inc.
    </small>
  </footer>
</div>
`;

exports[`<ErrorComponent /> renders correctly with link 1`] = `
<div
  className="dh-error"
>
  <header
    className="dh-error-header"
  >
    <div
      className="dh-error-logo"
    >
      <img
        alt=""
        src="http://placehold.it/150x30"
      />
    </div>
  </header>
  <section
    className="dh-error-section"
  >
    <div>
      <h1>
        Internal Server Error
      </h1>
      <p>
        We're sorry, but something went wrong. We have been notified about this and we'll take a look at it shortly.
      </p>
      <p>
        <a
          href="/"
        >
          Back to Homepage
        </a>
      </p>
    </div>
  </section>
  <footer
    className="dh-error-footer"
  >
    <small>
      © 
      2018
       
      Dering Hall, Inc.
    </small>
  </footer>
</div>
`;
